packages/frontend/src/ee/features/aiCopilot/components/AgentDetails.tsx

```diff
@@ -70,13 +88,36 @@ export const AgentDetails: FC = () => {
             void navigate('/generalSettings/aiAgents');
         },
     });
-    const isCreateMode = agentId === 'new';
-    const agentUuid = !isCreateMode && agentId ? agentId : undefined;
 
-    const { data: agent } = useAiAgent(agentUuid || '', {
-        enabled: !!agentUuid,
+    const [agentState, setAgentState] = useState<AgentState>(() => {
+        if (agentId === 'new') {
+            return {
+                mode: 'create',
+                data: null,
+                uuid: null,
+            };
+        }
+        return {
+            mode: 'loading',
+            data: null,
+            uuid: agentId || '',
+        };
+    });
+
+    const { data: agent } = useAiAgent(agentState.uuid || '', {
+        enabled: agentState.mode !== 'create' && !!agentState.uuid,
     });
 
+    useEffect(() => {
+        if (agentState.mode === 'loading' && agent) {
```

Comment:
I don't think we should have a `loading` `mode`
If it's loading, you should use: `useAiAgent`s `isLoading` property to track this.

That way you can remove this `useEffect`